code/game/machinery/transformer.dm

```diff
@@ -71,43 +76,45 @@
 		cooldown = FALSE
 		update_appearance()
 
-/obj/machinery/transformer/proc/do_transform(mob/living/carbon/human/H)
+/obj/machinery/transformer/proc/do_transform(mob/living/carbon/human/victim)
 	if(machine_stat & (BROKEN|NOPOWER))
 		return
-	if(cooldown == 1)
+
+	if(cooldown)
 		return
 
-	if(!transform_dead && H.stat == DEAD)
-		playsound(src.loc, 'sound/machines/buzz-sigh.ogg', 50, FALSE)
+	if(!transform_dead && victim.stat == DEAD)
+		playsound(src.loc, 'sound/machines/buzz-sigh.ogg', 5 SECONDS, FALSE)
 		return
 
 	// Activate the cooldown
-	cooldown = 1
+	cooldown = TRUE
 	cooldown_timer = world.time + cooldown_duration
 	update_appearance()
 
-	playsound(src.loc, 'sound/items/welder.ogg', 50, TRUE)
-	H.emote("scream") // It is painful
-	H.adjustBruteLoss(max(0, 80 - H.getBruteLoss())) // Hurt the human, don't try to kill them though.
+	playsound(src.loc, 'sound/items/welder.ogg', 5 SECONDS, TRUE)
+	victim.emote("scream") // It is painful
+	victim.adjustBruteLoss(max(0, 80 - victim.getBruteLoss())) // Hurt the human, don't try to kill them though.
 
 	// Sleep for a couple of ticks to allow the human to see the pain
 	sleep(0.5 SECONDS)
 
 	use_power(active_power_usage) // Use a lot of power.
-	var/mob/living/silicon/robot/R = H.Robotize()
-	R.cell = new /obj/item/stock_parts/cell/upgraded/plus(R, robot_cell_charge)
+	var/mob/living/silicon/robot/new_borg = victim.Robotize()
```

Comment:
Also, borgs generally get their AI connection during `Initialize()`. Overwriting it after creating the borg should work fine, except that if another AI exists (on the same Z-level), it'll be getting alerts for new borgs before they get re-assigned. 

Possibly, `Robotize()` should be updated to handle receiving a ref to the Malf AI, and then passing it along to the new borg's `Initialize()`.

Reply:
I was definitely thinking about doing this when I made the PR, I'll look into it!

Reply:
On second thought, leaving this for another PR.